billing/src/main/java/com/appcoins/wallet/billing/carrierbilling/CarrierResponseMapper.kt

```diff
@@ -54,6 +59,18 @@ class CarrierResponseMapper(private val retrofit: Retrofit) {
 
   private fun mapErrorResponseToCarrierError(httpCode: Int?,
                                              response: CarrierErrorResponse?): CarrierError? {
+    if (httpCode == FORBIDDEN_CODE) {
```

Comment:
I feel like this logic should be extracted since it is used on a lot of mappers but I'm not sure what's the best way to accomplish this. Maybe a "SubscriptionErrorMapper" that returns an enum of the error. The fact that we map to different error objects is also not very good, but I concede that.